### PR TITLE
Refactor http client and allow it on config

### PR DESCRIPTION
**Pull request by mullermp**
Pre-req for doing some client/connections in SRA - make http client a configurable option, and normalize the http option naming. Http client options will no longer be flat parameters on Config

### CODE REVIEW

File: hearth/lib/hearth/http/client.rb [hunk 1]
```diff
@@ -12,31 +12,31 @@ class Client
...
+      # @option options [String] :proxy A proxy to send
       #   requests through. Formatted like 'http://proxy.com:123'.
       #
-      # @option options [Boolean] :ssl_verify_peer (true) When `true`,
+      # @option options [Boolean] :verify_peer (true) When `true`,
       #   SSL peer certificates are verified when establishing a
       #   connection.
       #
-      # @option options [String] :ssl_ca_bundle Full path to the SSL
+      # @option options [String] :ca_file Full path to the SSL
```
Comment (alextwoods) [inline hearth/lib/hearth/http/client.rb:27]: the`ssl_` prefix feels a bit different than the `http` prefix - seems like these may still be useful in providing clarity on what its for?

File: hearth/lib/hearth/http/client.rb [hunk 2]
```diff
@@ -47,22 +47,21 @@ class Client
...
       end
 
       # @param [Request] request
       # @param [Response] response
       # @return [Response]
-      def transmit(request:, response:)
+      def transmit(request:, response:, **options)
         http = create_http(request.uri)
-        http.set_debug_output(@logger) if @http_wire_trace
+        http.set_debug_output(options[:logger]) if @debug_output
```
Comment (alextwoods) [inline hearth/lib/hearth/http/client.rb:64]: What happens here if logger is nil?

**Review by alextwoods — APPROVED** [review]

File: hearth/lib/hearth/http/client.rb (hunk 2 above)
Comment (mullermp) [reply hearth/lib/hearth/http/client.rb:64]: It shouldn't be, but I believe the method succeeds but no logging will occur.

File: hearth/lib/hearth/http/client.rb (hunk 1 above)
Comment (mullermp) [reply hearth/lib/hearth/http/client.rb:27]: These were changed to exactly what Net::HTTP uses. I think the prefixes made sense when they were public configuration. Now Client is the only public config.

**Review by alextwoods — APPROVED** [review]
Updates LGTM.

**Pull request merged**